src/RecoveryServices/RecoveryServices.SiteRecovery/ReplicationProtectedItem/NewAzureRmRecoveryServicesAsrReplicationProtectedItem.cs

```diff
@@ -729,9 +783,49 @@ private void AzureToAzureReplication(EnableProtectionInput input)
             }
 
 
+            providerSettings.DiskEncryptionInfo = this.A2AEncryptionDetailDetails();
+
             input.Properties.ProviderSpecificDetails = providerSettings;
         }
 
+        private DiskEncryptionInfo A2AEncryptionDetailDetails()
+        {
+            // Any encryption data is present.
+            if (this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionSecretUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.DiskEncryptionVaultId)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionKeyUrl)) ||
+                this.MyInvocation.BoundParameters.ContainsKey(Utilities.GetMemberName(() => this.KeyEncryptionVaultId)))
```

Comment:
You can just use `this.IsParameterBound(c => c.DiskEncryptionSecretUrl)`, etc. You used it about 300 lines up, but didn't use it for any of the logic below.